datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -30,151 +28,260 @@
+        return new AbstractSdxAction<>(SdxDetachFailedEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
+                    SdxDetachFailedEvent payload) {
                 return SdxContext.from(flowParameters, payload);
             }
 
             @Override
-            protected void doExecute(SdxContext context, SdxEvent payload, Map<Object, Object> variables) throws Exception {
-                LOGGER.info("SDX detaching in progress: {}", payload.getResourceId());
-                SdxCluster detachingSdxCluster = (SdxCluster) variables.get(DETACHING_SDX);
-                String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
-                LOGGER.info("Execute detach flow for SDX: {} Name: {}", payload.getResourceId(), detachingSdxCluster.getName());
-                ThreadBasedUserCrnProvider.doAsInternalActor(() ->
-                        stackV4Endpoint.updateNameAndCrn(0L,  (String) variables.get(SDX_NAME), initiatorUserCrn,
-                                detachingSdxCluster.getClusterName(), detachingSdxCluster.getCrn()));
-                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.STOPPED, "Data lake detaching in progress", payload.getResourceId());
-                if (variables.containsKey(RESIZED_SDX) && detachingSdxCluster.hasExternalDatabase()) {
-                    SdxCluster resizedSdxCluster = (SdxCluster) variables.get(RESIZED_SDX);
-                    LOGGER.info("Updating the cluster CRN from {} to {}", resizedSdxCluster.getCrn(), detachingSdxCluster.getCrn());
-                    // By now, CRN of the data lake being detached is updated to a new one and the new resized data lake object has the original CRN.
-                    // We need to find the DatabaseServerConfig entry using the original CRN and update it to use the new CRN used by the detached data lake.
-                    ThreadBasedUserCrnProvider.doAsInternalActor(() -> redbeamsServerEndpoint.updateClusterCrn(resizedSdxCluster.getEnvCrn(),
-                            resizedSdxCluster.getCrn(), detachingSdxCluster.getCrn(), initiatorUserCrn));
-                }
-                sendEvent(context, new SdxDetachSuccessEvent(SDX_DETACH_SUCCESS_EVENT.event(), payload.getResourceId(), payload.getUserId()));
+            protected void doExecute(SdxContext context, SdxDetachFailedEvent payload,
+                    Map<Object, Object> variables) throws Exception {
+                SdxCluster clusterToReattach = (SdxCluster) variables.get(DETACHED_SDX);
+                LOGGER.error("Failed to detach external DB of SDX with ID: {}. Attempting to restore it.", clusterToReattach.getId());
+
+                String detachedName = clusterToReattach.getClusterName();
+                clusterToReattach = sdxAttachService.reattachCluster(clusterToReattach);
```

Comment:
I still think you should introduce a new variable here instead of reusing the old, as at this point the cluster is reattached. but this is not a major thing

Reply:
Done.